src/proof/PGCheck.cc

```diff
@@ -21,102 +21,55 @@ along with Periplo. If not, see <http://www.gnu.org/licenses/>.
 #include "PG.h"
 
 #include "OsmtInternalException.h"
+#include "VerificationUtils.h"
 
 #include <deque>
-#include <sys/wait.h>
-#include <unistd.h>
 
 
 void ProofGraph::verifyLeavesInconsistency( )
 {
-	if( verbose() > 0 ) { std::cerr << "# Verifying unsatisfiability of the set of proof leaves" << std::endl; }
+    if (verbose() > 0) { std::cerr << "# Verifying unsatisfiability of the set of proof leaves" << std::endl; }
 
 	std::vector<clauseid_t> proofleaves;
-	clauseid_t id,id1,id2;
 	std::vector<clauseid_t> q;
-	ProofNode * node = NULL;
-	std::vector<unsigned>* visited_count_ = new std::vector<unsigned>(getGraphSize(),0);
-	std::vector<unsigned>& visited_count = *visited_count_;
-	// Building DFS vector
+	std::vector<unsigned> visited_count(getGraphSize(), 0u);
 	q.push_back(getRoot()->getId());
-	do
-	{
-		id = q.back();
-		node = getNode(id);
-		assert(node);
-		visited_count[id]++;
-		q.pop_back();
-
-		// All resolvents have been visited
-		if(id == getRoot()->getId() || visited_count[id] == node->getNumResolvents())
-		{
-			if(!node->isLeaf())
-			{
-				id1 = node->getAnt1()->getId();
-				id2 = node->getAnt2()->getId();
-				// Enqueue antecedents
-				assert( visited_count[id1] < node->getAnt1()->getNumResolvents() );
-				assert( visited_count[id2] < node->getAnt2()->getNumResolvents() );
-				q.push_back(id1); q.push_back(id2);
-			}
-			else proofleaves.push_back(id);
-		}
+    do {
+        clauseid_t id = q.back();
+        ProofNode * node = getNode(id);
+        assert(node);
+        visited_count[id]++;
+        q.pop_back();
+
+        // All resolvents have been visited
+        if (id == getRoot()->getId() or visited_count[id] == node->getNumResolvents()) {
+            if (not node->isLeaf()) {
+                clauseid_t id1 = node->getAnt1()->getId();
+                clauseid_t id2 = node->getAnt2()->getId();
+                // Enqueue antecedents
+                assert(visited_count[id1] < node->getAnt1()->getNumResolvents());
+                assert(visited_count[id2] < node->getAnt2()->getNumResolvents());
+                q.push_back(id1);
+                q.push_back(id2);
+            } else {
+                proofleaves.push_back(id);
+            }
+        }
+    } while (!q.empty());
+
+    vec<PTRef> clauses;
+	for (clauseid_t leaf_id : leaves_ids) {
+        auto const & clause = getNode(leaf_id)->getClause();
+        vec<PTRef> lits;
+        for (Lit l : clause) {
+            lits.push(termMapper.litToPTRef(l));
+        }
+        clauses.push(logic_.mkOr(std::move(lits)));
 	}
```

Comment:
This looks like an indentation situation.

Reply:
Fixed.